Analyticsv3Sample/AnalyticsSampleProject/StandardGoogleAuthSample.cs

```diff
@@ -0,0 +1,239 @@
+ 
+// Copyright 2015 DAIMTO :  www.daimto.com
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by DAIMTO-Google-apis-Sample-generator 1.0.0
+//     Build date: 11/17/2015 13:52:58
+//     C# generater version: 1.0.0
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Analytics.v3;
+using Google.Apis.Services;
+using Google.Apis.Util.Store;
+using System;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+/// <summary>
+/// API Name: Analytics.v3
+/// Description: View and manage your Google Analytics data
+/// Documentation: https://developers.google.com/analytics/
+/// Nuget Package
+/// https://www.nuget.org/packages/Google.Apis.Analytics.v3/
+/// PM> Install-Package Google.Apis.Analytics.v3 
+/// </summary>
+namespace DAIMTO.Analytics.v3.Sample
+{
+    public class StandardGoogleAuth
+    {
+
+	 /// <summary>
+        /// The Google APIs Client Library for .net uses the client_secrets.json file format for storing the client_id, client_secret, and other OAuth 2.0 parameters.
+        /// The client_secrets.json file format is a JSON formatted file containing the client ID, client secret, and other OAuth 2.0 parameters.
+        ///      
+        /// This file can be obtained from Google Developers console:   https://console.developers.google.com/project?authuser=0
+        /// </summary>
+        /// <param name="clientSecretJson">Path to the client secret json file from Google Developers console.</param>
+        /// <param name="userName">The user to authorize.</param>
+        /// <returns>a valid AnalyticsService</returns>
+        public static AnalyticsService AuthenticateOauth(string clientSecretJson, string userName)
```

Comment:
I have just added the methods i use most often.   We can create what ever we like.   

*  Lets start thinking about best practice as well.   
*  Do you think we should add a another with GetAuthenticatedServiceAsync.    
*  What will the users need? 
*  What do we want to show them?